### PR TITLE
Add -enable-gui option

### LINKED ISSUE #1318: Add -enable-gui option
This option will enable the serving of static GUI content.  Run it by default in run.sh and electron.

### PR DESCRIPTION
**Pull request by iketheadore**
Fixes #1318 

Changes:
- Add `-enable-gui-api` option

Does this change need to mentioned in CHANGELOG.md?
No

### CODE REVIEW

File: src/gui/http_test.go [hunk 1]
```diff
@@ -276,3 +277,54 @@ func TestGetBalanceHandler(t *testing.T) {
...
+
+	for _, tc := range tt {
+		t.Run(tc.name, func(t *testing.T) {
+			c := Config{
+				EnableGUI:   tc.enableGUI,
+				DisableCSRF: true,
+				StaticDir:   "./static",
+			}
+
+			host := "127.0.0.1:6423"
```
Comment (gz-c) [inline src/gui/http_test.go]: The mux method used in the other tests doesn't work here?

Comment (iketheadore) [reply src/gui/http_test.go]: Do you mean the `newServerMux` method?, it's created in the `Create` function.

File: src/gui/http.go [hunk 2]
```diff
@@ -83,7 +92,7 @@ func create(host string, c Config, daemon *daemon.Daemon) (*Server, error) {
 	if c.EnableJSON20RPC {
 		logger.Info("JSON 2.0 RPC enabled")
 		var err error
-		rpc, err = webrpc.New(daemon.Gateway)
+		rpc, err = webrpc.New(gateway.(*daemon.Gateway))
```
Comment (gz-c) [inline src/gui/http.go]: Add a note, `TODO -- change webrpc.New to use Gatewayer`

File: src/gui/http.go [hunk 3]
```diff
@@ -55,15 +56,23 @@ type Config struct {
...
-	enableWalletAPI bool
+	enableGUI       bool
 	enableJSON20RPC bool
 }
 
-func create(host string, c Config, daemon *daemon.Daemon) (*Server, error) {
-	var appLoc string
-	if c.EnableWalletAPI {
-		logger.Info("Wallet API enabled")
+// type StatusRespWriter struct {
```
Comment (gz-c) [inline src/gui/http.go]: What is this commented out?

**Pull request merged**